Source/Core/InputCommon/ControllerEmu/StickGate.cpp

```diff
@@ -25,18 +25,24 @@ constexpr auto CALIBRATION_CONFIG_SCALE = 100;
 constexpr auto CENTER_CONFIG_NAME = "Center";
 constexpr auto CENTER_CONFIG_SCALE = 100;
 
-// Calculate distance to intersection of a ray with a line defined by two points.
-double GetRayLineIntersection(Common::DVec2 ray, Common::DVec2 point1, Common::DVec2 point2)
+// Calculate distance to intersection of a ray with a line segment defined by two points.
+std::optional<double> GetRayLineIntersection(Common::DVec2 ray, Common::DVec2 point1,
```

Comment:
This might be useful to move to Math?

Reply:
I'd say move it to `Math` when it's needed in more than one place. Adding it to the math headers now would cause recompilation of basically everything for no benefit.